### PR TITLE
doc: add more links to Bentoml/examples

### PR DESCRIPTION
**Pull request by larme**
## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

**Comment by larme** [comment]
@aarnphm `hoverxref` seems only works on crossing reference so I did not include it in this pr (which mainly links to outside links)

File: docs/source/concepts/runner.rst [hunk 1]
```diff
@@ -70,8 +70,9 @@ Creating a Runnable
 
 Runner can be created from a :ref:`bentoml.Runnable <reference/core:bentoml.Runnable>`
 class. By implementing a :code:`Runnable` class, users can create Runner instances that
-runs custom logic. Here's an example, creating an NLTK runner that does sentiment
-analysis with a pre-trained model:
+runs custom logic. Here's an example (adopted from `one of our example projects
+<https://github.com/bentoml/BentoML/tree/main/examples/custom_runner/nltk_pretrained_model>`_)
+creating an NLTK runner that does sentiment analysis with a pre-trained model:
```
Comment (aarnphm) [inline docs/source/concepts/runner.rst:75]:
```suggestion
runs custom logic. Here's an example (excerpted from :examples:`one of our example projects
<custom_runner/nltk_pretrained_model>`)
creating an NLTK runner that does sentiment analysis with a pre-trained model:
```

can we add an extlinks to our `docs/source/conf.py`:

```python
extlink = {
"examples": ("https://github.com/bentoml/BentoML/tree/main/examples/%s", "%s"),  # noqa: WPS323
}

File: docs/source/concepts/runner.rst [hunk 2]
```diff
@@ -102,6 +103,8 @@ The :code:`bentoml.Runnable.method` decorator is used for creating
 for accessing remotely. :code:`RunnableMethod` can be configured with a signature,
 which is defined same as the :ref:`concepts/model:Model Signatures`.
 
+More examples about custom runners implementing their own :code:`Runnable` class can be found at:
+`Examples: Custom Runners <https://github.com/bentoml/BentoML/tree/main/examples/custom_runner>`_
```
Comment (aarnphm) [inline docs/source/concepts/runner.rst:107]: same as above

File: docs/source/concepts/runner.rst [hunk 3]
```diff
@@ -192,6 +195,12 @@ Custom Runnable built with Model from BentoML's model store:
         return {"res": spam_detection_runner.is_spam.run(input_text)}
 
 
+Custom Runnable can be also built by extending the Runnable class generated by Model
+from BentoML's model store. A full example with custom metrics to monitor the model's
+performance can be found at `Examples: Custom PyTorch MNIST runner
+<https://github.com/bentoml/BentoML/tree/main/examples/custom_model_runner>`_
```
Comment (aarnphm) [inline docs/source/concepts/runner.rst:201]: Same here.

File: docs/source/concepts/runner.rst [hunk 4]
```diff
@@ -274,6 +283,11 @@ In cases where certain steps can be executed concurrently, :code:`asyncio.gather
 Once each model completes, the results can be compared and logged as a post processing
 step.
 
+Another example generating text using three different language models concurrently and
+then classify each generated paragraph with an classification model can be found at:
+`Examples: Inference Graph Tutorial
```
Comment (aarnphm) [inline docs/source/concepts/runner.rst:288]: same

File: docs/source/concepts/service.rst [hunk 5]
```diff
@@ -307,7 +307,9 @@ JSON
 The data type of a JSON IO descriptor can be specified through a Pydantic model. By setting 
 a pydantic model, the IO descriptor will validate the input based on the specified pydantic
 model and return. To learn more, see IO descrptor reference for
-:ref:`reference/api_io_descriptors:Structured Data with JSON`.
+:ref:`reference/api_io_descriptors:Structured Data with JSON`. We also provide `an example
+project <https://github.com/bentoml/BentoML/tree/main/examples/pydantic_validation>`_
```
Comment (aarnphm) [inline docs/source/concepts/service.rst:311]: same here.

File: docs/source/frameworks/xgboost.rst [hunk 6]
```diff
@@ -178,3 +178,10 @@ default, but can be enabled using the appropriate signature when saving your boo
 .. code-block:: python
 
    bento_model = bentoml.xgboost.save_model("booster_tree", booster, signatures={"predict": {"batchable": True}})
+
+.. note::
+
+   You can find more examples for **XGBoost** in our :github:`bentoml/examples/xgboost
+   <https://github.com/bentoml/BentoML/tree/main/examples/xgboost>` directory.
```
Comment (aarnphm) [inline docs/source/frameworks/xgboost.rst:185]: same here

File: examples/custom_web_serving/flask_example/service.py [hunk 7]
```diff
@@ -26,7 +26,7 @@ def metadata():
     return {"name": bento_model.tag.name, "version": bento_model.tag.version}
 
 
-# For demo purpose, here's an identical inference endpoint implemented via FastAPI
+# For demo purpose, here's an identical inference endpoint implemented via Flask
```
Comment (aarnphm) [inline examples/custom_web_serving/flask_example/service.py:29]: Let's make a separate PR from this?

**Review by aarnphm — COMMENTED** [review]
Quick review of a few small things. Otherwise LGTM

**Comment by larme** [comment]
@aarnphm please look again

**Review by aarnphm — APPROVED** [review]
LGTM. Great work!!
